### DISCUSSION TITLE
"after_connect" option to run sql on successful connection

### DISCUSSION THREAD

**Discussion by HoneyryderChuck** [sqlalchemy/sqlalchemy#9089] [Usage Questions] [answered]
Hi,

I'm using SQLAlchemy 1.4 with psycopg2, (via `flask_sqlalchemy` integration= to connect to a postgres database behind an RDS proxy. I need to set timeouts on the connection. The easier option would be to use smth like `{ "connect_args": "-c statement_timeout=20s", but [RDS proxy does not allow me to do that](https://stackoverflow.com/questions/67242059/aws-rds-proxy-error-postgres-rds-proxy-currently-doesn-t-support-command-lin), and instead suggests me to set these up via SQL statement.

However, I don't find an option for create_engine documenting how to do it. Essentially I'd need smth like an after_connect callback I can enrich, like [ruby sequel has](https://github.com/jeremyevans/sequel/blob/master/doc/opening_databases.rdoc#general-connection-options-). Is there smth?

**Comment by zzzeek** [comment, marked as answer]
that would be [PoolEvents.connect](https://docs.sqlalchemy.org/en/14/core/events.html#sqlalchemy.events.PoolEvents.connect)

**Reply by HoneyryderChuck** [reply]
thx, it works!

To spare the next person on "how to execute a query":

```python
@event.listens_for(Pool, "connect")
def receive_connect(dbapi_connection, _connection_record):
    with dbapi_connection.cursor() as curs:
        curs.execute("SELECT ...
```
